src/query/impl/RTreeWrapper.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <map>
+#include "RTreeAdapter.hpp"
+
+namespace precice {
+
+namespace testing {
+namespace accessors {
+struct rtree;
+}
+} // namespace testing
+
+namespace query {
+namespace impl {
+
+using VertexTraits   = impl::RTreeTraits<mesh::Vertex>;
+using EdgeTraits     = impl::RTreeTraits<mesh::Edge>;
+using TriangleTraits = impl::RTreeTraits<mesh::Triangle>;
+
+struct MeshIndices {
+  VertexTraits::Ptr   vertexRTree;
+  EdgeTraits::Ptr     edgeRTree;
+  TriangleTraits::Ptr triangleRTree;
+};
+
+/// Class to encapsulate boost::geometry implementations
+class RTreeWrapper {
```

Comment:
The `RTreeWrapper` is a stateless proxyclass to the underlying `static map<>`. The user has to instantiate this, but there is really no object to speak of.

I see 2 possible ways to proceed:
1) we turn the class into a namespace, or
2) we  implement the `RTreeWrapper` as a singleton holding the map as a member and forcing access via some `static RTreeWrapper instance();`.

I would prefer the latter, as this will make things easier to understand:  
There is one instance of `<something>`. This instance is responsible for indexing meshes if required and caching them to prevent recomputations.
Given the above description, a more suitable name for the singleton would be `Indexer`.